### PR TITLE
Added HitRate metric with tests

### PR DESCRIPTION
**Pull request by Altair7610**
## Description

Added HitRate metric 

Closes https://github.com/MobileTeleSystems/RecTools/issues/94

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Optimization

## How Has This Been Tested?
Before submitting a PR, please check yourself against the following list. It would save us quite a lot of time.
- Have you read the contribution guide? - Yes
- Have you updated the relevant docstrings? We're using Numpy format, please double-check yourself - Yes
- Does your change require any new tests? - Already written
- Have you updated the changelog file? - Yes

### CODE REVIEW

File: rectools/metrics/classification.py [hunk 1]
```diff
@@ -371,6 +371,27 @@ def _calc_per_user_from_confusion_df(self, confusion_df: pd.DataFrame, catalog:
         return mcc
 
 
+@attr.s
+class HitRate(SimpleClassificationMetric):
+    """
+    HitRate calculates the fraction of users for which the correct answer is included in the recommendation list
+
+    The HitRate equals to ``tp / (tp + fn)`` where
```
Comment (feldlime) [inline rectools/metrics/classification.py:379]: Hi! Thank you for the PR
However this formula is for `Recall`, not for `HitRate`.

P.S. If your PR is ready, please add reviewers (me or @blondered). If not, please mark it as draft

Comment (Altair7610) [reply rectools/metrics/classification.py]: Sorry, when I read the artcile about metrics, the wrong calculation method stuck in my head.. Fixed

File: tests/metrics/test_classification.py [hunk 2]
```diff
@@ -155,3 +155,30 @@ def test_when_no_interactions(self) -> None:
...
+    def test_calc(self) -> None:
+
+        # tp = pd.Series([1, 1, 0, 0])
+        # tn = pd.Series([6, 8, 7, 7])
+        # fp = pd.Series([1, 1, 2, 2])
+        # fn = pd.Series([2, 0, 1, 1])
+
+        expected_metric_per_user = pd.Series(
+            [1, 1, 0, 0],
+            index=pd.Series([1, 3, 4, 5], name=Columns.User, dtype=int), dtype=float
```
Comment (feldlime) [inline tests/metrics/test_classification.py:173]: Please execute `make lint` and `make test` before pushing the code to be sure everything is fine.  
To format the code automatically, run `make format`

[the remark this answers is not in the document]
Comment (Altair7610) [reply tests/metrics/test_classification.py]: A-OK, there are some test errors in dssm, lightfm, torch dataset and visual app and lint errors in dssm, but I believe they were already there

File: rectools/metrics/classification.py (hunk 1 above)
Comment (feldlime) [reply rectools/metrics/classification.py]: That's ok

For future: please do not resolve the threads yourself, the thread must be resolved by its creator

File: tests/metrics/test_classification.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (feldlime) [reply tests/metrics/test_classification.py]: It seems there are no mistakes now

File: rectools/metrics/classification.py [hunk 3]
```diff
@@ -371,6 +371,26 @@ def _calc_per_user_from_confusion_df(self, confusion_df: pd.DataFrame, catalog:
         return mcc
 
 
+@attr.s
+class HitRate(SimpleClassificationMetric):
+    """
+    HitRate calculates the fraction of users for which the correct answer is included in the recommendation list
```
Comment (feldlime) [inline rectools/metrics/classification.py:377]: There should be a dot at the end of the sentence.

File: rectools/metrics/classification.py [hunk 4]
```diff
@@ -371,6 +371,26 @@ def _calc_per_user_from_confusion_df(self, confusion_df: pd.DataFrame, catalog:
...
+
+    The HitRate equals to ``1 if tp > 0, otherwise 0`` where
+        - ``tp`` is the number of relevant recommendations
+          among the first ``k`` items in recommendation list;
```
Comment (feldlime) [inline rectools/metrics/classification.py:381]: Also a dot here

File: tests/metrics/test_classification.py [hunk 5]
```diff
@@ -155,3 +155,29 @@ def test_when_no_interactions(self) -> None:
             expected_metric_per_user,
         )
         assert np.isnan(self.metric.calc(RECO, EMPTY_INTERACTIONS, CATALOG))
+
+
+class TestHitRate:
```
Comment (feldlime) [inline tests/metrics/test_classification.py:160]: Please also add this metric to the complex test in `test_scoring.py`

File: rectools/metrics/classification.py
Comment (feldlime) [inline rectools/metrics/classification.py:1]: Please also update the Changelog

File: rectools/metrics/__init__.py [hunk 6]
```diff
@@ -72,4 +73,5 @@
     "PairwiseDistanceCalculator",
     "PairwiseHammingDistanceCalculator",
     "SparsePairwiseHammingDistanceCalculator",
+    "HitRate",
```
Comment (feldlime) [inline rectools/metrics/__init__.py:76]: It's better to put this metric close to other classification metrics, after the `Accuracy` e.g.

File: rectools/metrics/classification.py (hunk 3 above)
Comment (Altair7610) [reply rectools/metrics/classification.py]: Fix type here

File: rectools/metrics/classification.py (hunk 4 above)
Comment (Altair7610) [reply rectools/metrics/classification.py]: And fix typo here

File: tests/metrics/test_classification.py (hunk 5 above)
Comment (Altair7610) [reply tests/metrics/test_classification.py:160]: Metric landed here

File: rectools/metrics/classification.py
Comment (Altair7610) [reply rectools/metrics/classification.py:1]: Add my PR to changelog

File: rectools/metrics/__init__.py (hunk 6 above)
Comment (Altair7610) [reply rectools/metrics/__init__.py]: Moved up to classification ones

**Review by feldlime — APPROVED** [review]

**Pull request merged**